api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+        XWikiContext context = xwikiContextProvider.get();
+
+        if (context != null) {
+            readConfigDoc(context);
+        }
+
+        if (xwiki != null && context != null) {
+            log.info("Initting authService.");
+            // We do not verify with the context if the plugin is active and if the license is active
+            // this will be done by the GoogleAppsAuthService and UI pages later on, when it is called within a request
+            try {
+                authService = componentManager.getInstance(GoogleAppsAuthService.class);
+                authService.setGoogleAppsManager(this);
+                xwiki.setAuthService(authService);
+                log.info("Succeeded initting authService,");
+            } catch (ComponentLookupException e) {
+                log.info("Failed initting authService", e);
+            }
+        }
+        if (authService == null) {
+            log.info("Not yet initting authService.");
+        }
+
+        try {
+            jacksonFactory = JacksonFactory.getDefaultInstance();
+            httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (Exception  e) {
+            e.printStackTrace();
+            throw new InitializationException("Trouble at initializing", e);
+        }
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        return Arrays.asList(new ApplicationReadyEvent(), new DocumentUpdatedEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        log.info("Event triggered: " + event + " with source " + source);
+        boolean applicationStarted = false, configChanged = false;
+        if (event instanceof ApplicationReadyEvent) {
+            applicationStarted = true;
+        }
+        if (event instanceof DocumentUpdatedEvent) {
+            XWikiDocument document = (XWikiDocument) source;
+            configDocRef = getConfigDocRef();
+            if (document != null && document.getDocumentReference().compareTo(configDocRef) == 0) {
+                configChanged = true;
+            }
+        }
+
+        if(configChanged) {
+            log.info("Reloading config.");
+            readConfigDoc(xwikiContextProvider.get());
+        }
+
+        if(applicationStarted || configChanged) {
+            try {
+                initialize();
+            } catch (InitializationException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    // internals
+
+    private GoogleAppsAuthServiceImpl authService;
+
+    private DocumentReference configDocRef;
+    private ObjectReference configObjRef;
+
+    /** A map of hash to full redirects. */
+    private Map<String, String> storedStates = new HashMap<>();
+
+    private FileDataStoreFactory dsFactory;
+
+    private JacksonFactory jacksonFactory;
+
+    private NetHttpTransport httpTransport;
+
+    private CloseableHttpClient httpclient = HttpClients.createDefault();
+
```

Comment:
All of these declarations should go at the beginning of the class. With some check-style enabled, I think the build will fail.

Reply:
I am reluctant to do this now and prefer to do this when we refactor and then enable checkstyle. For now it is disabled.